From f0067b48e30e1ce07d2e6a95136ab0376ff18e9b Mon Sep 17 00:00:00 2001
From: Ilya Dryomov <ilya.dryomov@inktank.com>
Date: Fri, 20 Jun 2014 18:29:20 +0400
Subject: [PATCH 130/213] libceph: fix linger request check in
 __unregister_request()
References: fate#318918
Git-commit: 4f23409e0c787be1a501527c79578833567a1e58
Patch-mainline: v3.17-rc1

We should check if request is on the linger request list of any of the
OSDs, not whether request is registered or not.

Signed-off-by: Ilya Dryomov <ilya.dryomov@inktank.com>
Reviewed-by: Alex Elder <elder@linaro.org>
Acked-by: David Disseldorp <ddiss@suse.de>

---
 net/ceph/osd_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/ceph/osd_client.c b/net/ceph/osd_client.c
index bf48e58..87d5b49 100644
--- a/net/ceph/osd_client.c
+++ b/net/ceph/osd_client.c
@@ -1230,7 +1230,7 @@ static void __unregister_request(struct ceph_osd_client *osdc,
 
 		list_del_init(&req->r_osd_item);
 		maybe_move_osd_to_lru(osdc, req->r_osd);
-		if (list_empty(&req->r_linger_item))
+		if (list_empty(&req->r_linger_osd_item))
 			req->r_osd = NULL;
 	}
 
-- 
2.1.4

